js/src/Figure.ts

```diff
@@ -922,13 +929,23 @@ export class Figure extends widgets.DOMWidgetView {
     }
 
     _update_gl() {
-        this.render_gl()
         this._update_requested = false;
+        this.render_gl();
     }
 
-    render_gl() {
-        if(this.mark_views === undefined)
-            this.update_gl() // we got call to soon, maybe next frame?
+    render_gl() : Promise<void> {
+        // Nothing to render using a WebGL context
+        if (!this.renderer) {
+            return Promise.resolve();
+        }
+
+        // Marks are not created yet, maybe next frame?
+        if(this.mark_views === undefined) {
+            return Promise.resolve().then(() => {
+                this.update_gl();
```

Comment:
I think this will not resolve after the update_gl really did its work

Reply:
I actually think it does resolve the issue... Because at this point in the code `_update_requested` is false. So we actually call `requestAnimationFrame(() => this._update_gl());` inside `update_gl`, which was not the case before.

Reply:
> I think this will not resolve

Oh I thought you were talking about resolving the issue. You mean resolve the Promise?